src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs

```diff
@@ -431,6 +430,15 @@ public async Task<AuthFlowResponse> RespondToMfaAuthAsync(RespondToMfaRequest mf
         /// if one exists</returns>
         public async Task<AuthFlowResponse> RespondToMfaAuthAsync(RespondToMfaRequest mfaRequest, CancellationToken cancellationToken)
         {
+            if (mfaRequest == null)
+            {
+                throw new ArgumentNullException(nameof(mfaRequest));
+            }
```

Comment:
I believe this is a breaking change. Do you require this check?

Reply:
I don't require this check. Whithout this check NullReferenceException will be thrown at [ChallengeNameType property access](https://github.com/DmitryProskurin/aws-sdk-net-extensions-cognito/blob/c166c6a9608268543cf43d71c134896fed0c8aa5/src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs#L446C66-L446C66). This check help to provide more specific info about error, when pass null at mfaRequest parameter.
I doubt anyone has use case of handling an NullReferenceException when calling RespondToMfaAuthAsync method, and trying to determine that it occurred because of a passed null parameter. But for new users it provide exact reason of error.
If you think this is critical, then I can remove this check.

Reply:
You can keep it since as you mentioned, the error will eventually occur on line 437